common-npm-packages/artifacts-common/credentialProviderUtils.ts

```diff
@@ -155,6 +172,42 @@ export function configureCredProviderForServiceConnectionFeeds(serviceConnection
     }
 }
 
+/**
+ * Configure the credential provider to provide credentials for a single service connections,
+ * using VSS_NUGET_EXTERNAL_FEED_ENDPOINTS to do so.
+ */
+export function configureSingleCredProviderForServiceConnectionFeeds(serviceConnections: ServiceConnection[]) {
+    var configuredEndpoints = tl.getVariable(CRED_PROVIDER_EXTERNAL_ENDPOINTS_ENVVAR);
+
+    if (serviceConnections) {
+        console.log(tl.loc('CredProvider_SettingUpForServiceConnections'));
+        
+        serviceConnections.map(authInfo => `${authInfo.packageSource.uri}`).forEach(serviceConnectionUri => console.log('  ' + serviceConnectionUri))
```

Comment:
Note, this will work for the current service connection types, but for the new types there is no authInfo.packageSource.Uri so the add single will pass the service connection and feed url separately.

Reply:
this is fine for now, but consider generalizing so we dont have to update in the future

Reply:
Made changes to support the new service connection and (hopefully) large avoid future updates